daemon/logger/plaintextlog/plaintextlog.go

```diff
@@ -0,0 +1,181 @@
+package plaintextlog
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/daemon/logger"
+	"github.com/docker/docker/pkg/units"
+)
+
+const (
+	driverName = "plain-text"
+)
+
+type plainTextLogger struct {
+	buf      *bytes.Buffer
+	f        *os.File   // store for closing
+	mu       sync.Mutex // protects buffer
+	capacity int64      //maximum size of each file
+	n        int        //maximum number of files
+	ctx      logger.Context
+}
+
+func init() {
+	if err := logger.RegisterLogDriver(driverName, New); err != nil {
+		logrus.Fatal(err)
+	}
+	if err := logger.RegisterLogOptValidator(driverName, ValidateLogOpt); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+// New creates new plainTextLogger which writes to filename passed in
+// on given context.
+func New(ctx logger.Context) (logger.Logger, error) {
+	log, err := os.OpenFile(ctx.Config["log-path"], os.O_RDWR|os.O_APPEND|os.O_CREATE, 0600)
```

Comment:
Why `O_RDWR`? I don't see any code which reads from the file.

Also, this line is rather different from the one down on line 99 when you re-open the file after rotation. The flags are different (that one is `O_WRONLY`), as well as the permissions.